### PR TITLE
ICM-634. Added prefix "zipkin" for all request in UI.

### PR DESCRIPTION
**Pull request by knikitin**
[ICM-634. Service requests tracing.](https://intapp.atlassian.net/browse/ICM-634)

- Added prefix "zipkin".

### CODE REVIEW

File: zipkin-ui/js/component_data/default.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ export default component(function DefaultData() {
     const query = convertToApiQuery(window.location.search);
     const serviceName = query.serviceName;
     if (serviceName) {
-      const apiURL = `/api/v1/traces?${queryString.stringify(query)}`;
+      const apiURL = `/zipkin/api/v1/traces?${queryString.stringify(query)}`;
```
Comment (codefromthecrypt) [inline zipkin-ui/js/component_data/default.js:24]: is this only affecting crossroads? I don't expect us to mount the api under the subpath /zipkin at the server level by default.

Comment (codefromthecrypt) [reply zipkin-ui/js/component_data/default.js:24]: well at least not without significantly broadening the discussion beyond those interested in this UI (as it has deployment and other doc implications)

**Review by codefromthecrypt — COMMENTED** [review]
Thanks for the help!

:) good sign that there's no unit tests is that this passed without changing them. This means we have a bit more burden to figure out if this will work or not.

Also worth raising an issue or otherwise trying here as it is expected that this will finish at some point https://github.com/openzipkin/zipkin-ui

Finally, if you don't mind, there have been several issues here to link to on the same topic. mind pinging them on your work?

File: zipkin-ui/js/component_data/default.js (hunk 1 above)
Comment (eirslett) [reply zipkin-ui/js/component_data/default.js:24]: This would affect the API as well.

**Comment by codefromthecrypt** [comment]
is it possible to read this prefix from config.json (and parameterize all
other paths based on that value)? It would be nice to not have to play
games for the base case where the url space is root.

**Comment by eirslett** [comment]
> is it possible to read this prefix from config.json

There's a chicken-and-egg problem there, as the app doesn't know the location of the config.json file (today we assume it's located at root)

We could try to heuristically detect whether we're mounted under root or not:

```javascript
// inside the web UI code:

// check if pathname starts with /zipkin/, and if it does,
// assume that we're mounted under a /zipkin/ context path
const isMountedAtRoot = window.location.pathname.indexOf('/zipkin/') !== 0;
const basePath = isMountedAtRoot ? '/' : '/zipkin/';
```

That is assuming we don't have any request paths in our application that start with /zipkin/, but I don't think we have.

We would then support two use cases: 1) mounting zipkin on root, as we do today, and 2) mounting zipkin under /zipkin/.
The case we would not support is 3), to mount zipkin under an arbitrary context path, but that would be an edge case. I don't think we should take on the burden of supporting it. (Why would somebody want to mount an app under /zipkin/ that is not zipkin?)

The context path mapping itself can either be done on the zipkin server side if we choose to implement it (MOUNT_CONTEXT_PATH=true) or in a proxy setup (apache 2 or nginx) running in front of the server. In the latter case, this issue can be solved purely on the UI side, with no server side changes necessary. People who mount apps under a context path, with multiple apps on the same domain, are already using a frontend proxy anyways.

**Comment by codefromthecrypt** [comment]
I like this idea

File: zipkin-ui/js/component_data/default.js (hunk 1 above)
Comment (knikitin) [reply zipkin-ui/js/component_data/default.js:24]: Sorry. I made a mistake with the creation of the pull request. I made a mistake with the choice of destination.  
If you do not mind, I will close this pull request.

**Comment by tschiman** [comment]
Sorry to drag this thread back to life, but did this get added to the project? 

I see many threads in the issues about this topic, and I am another user that would like to see this feature added. More specifically the one where I can set the prefix myself. If this isn't the most active thread on this topic can you point me to it?

**Pull request closed without merging**